library/java/src/io/envoyproxy/envoymobile/engine/JvmCallbackContext.java

```diff
@@ -167,14 +181,16 @@ public void run() {
    * Cancel the callback context atomically so that no further callbacks occur
    * other than onCancel.
    *
-   * @return boolean, whether the callback context was canceled or not.
+   * @return boolean, whether the callback context was closed or not.
    */
   public boolean cancel() {
-    boolean canceled = !this.canceled.getAndSet(true);
-    if (canceled) {
+    // Atomically close the stream if not already closed.
+    boolean closed = dispatchable(true);
+    if (closed) {
+      // Directly fire callback if closure occurred.
```

Comment:
can you remind me why we don't reset the stream here?

Reply:
https://github.com/lyft/envoy-mobile/blob/ms/java-termcbs/library/java/src/io/envoyproxy/envoymobile/engine/EnvoyHTTPStream.java#L80

Reply:
(but thank you for helping me catch a bug)